rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+
+**Decision**: CommonMark is the content of all doc-comments.
+
+> Markdown is the most accepted and understood format for writing documentation in tech projects. Also, following the existing RFC-72 is highly profitable and consistent for the Nix ecosystem.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) CommonMark is the official format in nix; Decided in [RFC72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+    - (+) It Would be consistent if this RFC builds upon the previous one.
+    - (+) Further Arguments for CommonMark, in general, can be found in RFC72
+- (+) Allows copy-paste from and to markdown files. We allow easy refactoring if documentation arises and needs to be split into separate files.
+- (-) Strictly binding doc comments content to commonMark might restrict users.
+    - (+) Users/Tools can still use regular comments or develop alternative solutions.
+- (-) CommonMark does not specify the current rich features from nixdoc, such as predefined sections and structures that could be used as a source for automated toolings. Such as types for type-checking or examples to run automated-tests
+    - (+) Future tools can still build their conventions on top of this RFC. They might not directly specify them in an RFC but be a tool developers choose for a specific codebase. However, we have yet to get that tools. So it is good when rich features remain unspecified.
+  
+</details>
+
+## Single-line doc-comments (do not exist)
+
+**Observing**: Nix offers two variants of comments; single- (`#`) and multi-line comments (`/* */`). There may be use cases where it is desirable to have a form of single-line comments subtyped for doc-comment purposes.
+
+**Considering**: Single-line comment for documentation. (Starting a doc-comment with `##`)
+
+**Decision**: Single-line comments (starting with `##`) **cannot be used** in any form for documentation puposes.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) It Would be consistent with providing variants for both nix comments.
+- (-) Doc-comments should have only one variant to reduce complexity.  
+- (-) documentation will likely everytime take up more than one line.
+- (-) If documentation grows bigger than one line, refactoring into a multiline-doc-comment must occur.
+- (+) Offer the choice.
+- (o) Single lines could also be concatenated to form multi-line documentation.
+  - (+) Convinience
+  - (-) Technical more complex
+- (+) Takes up less vertical space
+- (-) Visually confusing when every line starts with a `#` character.
+    - (-) Potential visual conflicts with the content that is markdown (e.g. with headings).
+- (+) Indentation of the content is clear.
+
+</details>
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Changes the existing comments inside the code base
+
+This could be automated.
+
+Also, the migration could be performed piecemal, starting perhaps with `nixpkgs.lib`.
+
+See future work.
+
+## Tooling to produce the nixpkgs manual
```

Comment:
What's the drawback?

Reply:
What do you mean? We should talk with somebody familiar with the current manual and make sure that it adds support for markdown. 
future work? 
A sentence like:
„
All current rendering solutions should support rendering markdown from the doc-comment. „

Reply:
I mean that we're in the drawbacks section, and it's not clear from the title what the drawback is.